client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,197 @@
+<script lang="ts">
+  import type { PrivatePlayer, Lobby } from "../../../functions/src/firestore-types/lobby";
+  import { onMount } from "svelte";
+  import { lobbyReturn } from "$lib/firebase/firestore-functions";
+
+  export let lobbyCode: string;
+  export let lobby: Lobby;
+  export let privatePlayer: PrivatePlayer;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishes: string[];
+  onMount(() => {
+    // format a string list to have commas and "and" when needed
+    const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+    // get the list of catfishes from lobby.ts and put them into a variable
+    catfishes = lobby.players
+      .filter((p) => {
+        return p.role == "CATFISH";
+      })
+      .map((c) => c.displayName);
+    catfishList = formatter.format(catfishes);
+
+    // grab the current user's role
+    const myRole = privatePlayer.role;
+    // update end variable depending on the winner type and the current player's role
+    if (myRole == "CAT") {
+      if (lobby.winner == "CAT") {
+        end = "Cat Win";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Cat Lose";
+      }
+    } else {
+      if (lobby.winner == "CAT") {
+        end = "Catfish Lose";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Catfish Win";
+      }
+    }
+  });
+</script>
+
+<div class="container">
+  <!-- Check the user's role and display the correct win or loss page accordingly -->
+  {#if end == "Cat Win"}
+    <div class="banner">
+      <h2>Hooray!</h2>
+      <h2>You have sniffed out the cat fish!</h2>
+    </div>
+
+    <div class="image">
+      <img src="https://www.azpetvet.com/wp-content/uploads/Happy-cat-1024x711.jpg" alt="clown fish" />
```

Comment:
do we have rights to this image?

Reply:
I'm just kind of using these images as placeholders for this sprint. I'm not entirely sure if we've come to a decision on what we want here yet

Reply:
I'm just kind of using these images as placeholders for this sprint. I'm not entirely sure if we've come to a decision on what we want here yet

Reply:
I went ahead and just drew some pictures in paint so let me know how you feel about them